From 8b3ecdf1f4f120878a43da89afeed62533acb6c4 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 19 Apr 2013 11:22:32 +0200
Subject: [PATCH] abuild: add support for optional remote logging

via ABUILD_LOG_CMD config option

This is supposed to be used on the build servers
---
 abuild.in | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/abuild.in b/abuild.in
index 284ee6f..76a0806 100755
--- a/abuild.in
+++ b/abuild.in
@@ -5,7 +5,7 @@
 #
 # Distributed under GPL-2
 #
-# Depends on: busybox utilities, fakeroot, 
+# Depends on: busybox utilities, fakeroot
 #
 
 abuild_ver=@VERSION@
@@ -52,8 +52,13 @@ monochrome() {
 #colors
 if [ -n "$USE_COLORS" ]; then
     default_colors
-fi	
-	
+fi
+
+# run optional log command for remote logging
+logcmd() {
+	${ABUILD_LOG_CMD:-true} "$@"
+	return 0
+}
 
 # functions
 msg() {
@@ -87,6 +92,7 @@ error() {
 	local fake="${FAKEROOTKEY:+${BLUE}*${NORMAL}}"
 	local name="${STRONG}${subpkgname:-$pkgname}${NORMAL}"
 	printf "${prompt} ${name}${fake}: %s\n" "$1" >&2
+	logcmd "ERROR: $pkgname: $1"
 }
 
 error2() {
@@ -1161,6 +1167,7 @@ mklinks_abuildrepo() {
 build_abuildrepo() {
 	local d apk
 	if ! apk_up2date || [ -n "$force" ]; then
+		logcmd "building $pkgname"
 		sanitycheck && builddeps && clean && fetch && unpack \
 			&& prepare && mkusers && rootpkg || return 1
 	fi
-- 
1.8.2.1

